### PR TITLE
symintify rand and randint functions and meta suport for randint

### PR DESCRIPTION
**Pull request by anjali411**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #86358

### CODE REVIEW

File: aten/src/ATen/functorch/BatchRulesRandomness.cpp [hunk 1]
```diff
@@ -18,10 +18,10 @@ namespace at {
 namespace functorch {
 
 template <typename F, F Func, typename... ExtraArgs>
-Tensor random_batching_rule(IntArrayRef shape, ExtraArgs... extra_args) {
+Tensor random_batching_rule(SymIntArrayRef shape, ExtraArgs... extra_args) {
   c10::impl::ExcludeDispatchKeyGuard guard(DispatchKey::FuncTorchVmapMode);
   auto maybe_layer = maybeCurrentDynamicLayer();
-  VmapDimVector shapeVec(1, maybe_layer->batchSize());
+  c10::SmallVector<SymInt> shapeVec(1, maybe_layer->batchSize());
```
Comment (albanD) [inline aten/src/ATen/functorch/BatchRulesRandomness.cpp:24]: Not sure if this is ok given @ezyang 's latest ambitions?

File: torch/_meta_registrations.py [hunk 2]
```diff
@@ -80,6 +80,9 @@ def meta_randperm(n, *, generator=None, out):
     assert out.ndim == 1 and out.size(0) == n
     return out
 
+@register_meta(aten.randint.default)
+def meta_randperm(high, size, dtype, layout, device, pin_memory):
+    return torch.empty(size, dtype=dtype, device=device)
```
Comment (albanD) [inline torch/_meta_registrations.py:85]: Why not pass layout and pin memory?

**Review by albanD — COMMENTED** [review]
Sounds ok to me but would like a double check from Ed

File: torch/_meta_registrations.py (hunk 2 above)
Comment (anjali411) [reply torch/_meta_registrations.py:85]: oh yeah that needs to be updated

File: aten/src/ATen/functorch/BatchRulesRandomness.cpp (hunk 1 above)
Comment (ezyang) [reply aten/src/ATen/functorch/BatchRulesRandomness.cpp:24]: Use SymDimVector instead, but this is fine, I gave up on changing SymIntArrayRef

File: torch/_meta_registrations.py (hunk 2 above)
Comment (ezyang) [reply torch/_meta_registrations.py:85]: one idiom for factory functions is to just pass along the entire `**kwargs`

**Review by ezyang — APPROVED** [review]

**Review by albanD — APPROVED** [review]
Just the kwarg update otherwise LGTM !

**Pull request closed without merging**